### ISSUE TITLE
fio-postprocess ignores multiple jobs

### ISSUE THREAD

**Issue by k-rister** [distributed-system-analysis/pbench#740]
https://github.com/distributed-system-analysis/pbench/blob/2cfe1306d66604dda13b1256471dac83a0dec368/agent/bench-scripts/postprocess/fio-postprocess#L175

When pbench-fio is given multiple targets it generates multiple jobs in the fio jobfile.  The results in files being generated with the form (fio_bw|fio_clat|fio_clat_hist|fio_iops|fio_lat|fio_slat).N.log where N is the job number (starting at 1).  As shown above, fio-postprocess only processes the first job (N=1) and ignores all others.

For example, the following command:

`pbench-fio --config=foo --test-types=write --ramptime=30 --direct=1 --runtime=120 --block-sizes=4 --targets=/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350029400BGN,/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350059400BGN --numjobs=1 --iodepth=32 --samples=1`

Generates the following fio jobfile:

```
[global]
bs=4k
runtime=120
ioengine=libaio
iodepth=32
direct=1
sync=0
time_based=1
clocksource=gettimeofday
ramp_time=5

[job-/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350029400BGN]
filename=/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350029400BGN
rw=write
size=4096M
write_bw_log=fio
write_iops_log=fio
write_lat_log=fio
log_avg_msec=1000
write_hist_log=fio
log_hist_msec=10000
log_hist_coarseness=4 # 76 bins

[job-/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350059400BGN]
filename=/dev/disk/by-id/nvme-INTEL_SSDPEDMD400G4_CVFT52350059400BGN
rw=write
size=4096M
write_bw_log=fio
write_iops_log=fio
write_lat_log=fio
log_avg_msec=1000
write_hist_log=fio
log_hist_msec=10000
log_hist_coarseness=4 # 76 bins
```

Which when run results in the following log files being created:

![image](https://user-images.githubusercontent.com/15911116/34794436-3d9ea0c6-f614-11e7-99b4-2c19339d9568.png)

It is also possible to encounter this problem when the user supplies their own fio jobfile.  In that situation, there may be any number of jobs created.  In either case, only the first job is processed into the resulting JSON and summary files.

**Comment by k-rister** [comment]
@atheurer 

FYI, I dug a little bit deeper on what we discussed and given what I found I filed this issue

**Comment by k-rister** [comment]
I've come up with a simple fix that forces jobs to be treated as individual clients.  I'm not sure if it is the best approach but it is a small change that results in all the results being properly processed.

Before:

![image](https://user-images.githubusercontent.com/15911116/35227248-2c69deb8-ff53-11e7-9ba3-9a2cfd29506d.png)

After:

![image](https://user-images.githubusercontent.com/15911116/35227282-4cd8c81c-ff53-11e7-972a-42af1f6f6ad0.png)

[Link to the proposed patch](https://github.com/k-rister/pbench/commit/1af735f0d5b1a1b07bb153edf0eac30e8ad90e79)

**Comment by k-rister** [comment]
Addressed by #761

**Comment by k-rister** [comment]
PR #761 was merged which fixes this issue so closing.
